### ISSUE TITLE
HTML not getting decoded properly by Middleware in Django 2.2

### ISSUE THREAD

**Issue by ErrorInPersona** [jazzband/django-pipeline#691]
Hey!
Thanks for the great repo. I´m experiencing an issue with the MinifyHTMLMiddleware. When active, my HTML-Output looks like b'...\n\n'. So it is a bytes-literal.
I can fix it by changing 
`response.content = minify_html(response.content.strip())`
(https://github.com/jazzband/django-pipeline/blob/master/pipeline/middleware.py#L25)
to:
`response.content = minify_html(response.content.decode().strip())`

Just so you know about the issue.
Kind regards! :)

**Comment by hugotacito** [comment]
I'm experiencing the same issue. Your solution fixed the problem here for me as well. Anyone can fix this on the project?

**Comment by gatsinski** [comment]
The fix suggested by @ErrorInPersona really solves the issue and works fine on 1.11, 2.1 and 2.2.

**Comment by takis** [comment]
I'm also experiencing the same issue, on Django 2.2.12 and django-pipeline 1.6.14 and 1.7.0. The suggested fix works for me.

**Comment by TheBuky** [comment]
Do you always have this problem with version 2.0?

Else I will merge #706 and try to add test to avoid having this error again

**Comment by malefice** [comment]
I tested using Django 2.0.13, and it does not have this problem, but 2.1 and onward will have this problem. This is caused by the change in `strip_spaces_between_tags` (aliased as `minify_html` in pipeline's middleware) between [2.0](https://github.com/django/django/blob/stable/2.0.x/django/utils/html.py#L181) and [2.1](https://github.com/django/django/blob/stable/2.1.x/django/utils/html.py#L199).

For version 2.0, `django.utils.encoding.force_text` was being used and the response bytes object is being decoded properly under the hood. For 2.1 and above, `force_text` is no longer being used, and the response bytes object is being passed to `str` directly without the `encoding` keyword argument, so no decoding takes place.

**TL;DR:** The middleware should handle the decoding which the PR does.

**Comment by ErrorInPersona** [comment]
> 
> 
> I tested using Django 2.0.13, and it does not have this problem, but 2.1 and onward will have this problem. This is caused by the change in `strip_spaces_between_tags` (aliased as `minify_html` in pipeline's middleware) between [2.0](https://github.com/django/django/blob/stable/2.0.x/django/utils/html.py#L181) and [2.1](https://github.com/django/django/blob/stable/2.1.x/django/utils/html.py#L199).
> 
> For version 2.0, `django.utils.encoding.force_text` was being used and the response bytes object is being decoded properly under the hood. For 2.1 and above, `force_text` is no longer being used, and the response bytes object is being passed to `str` directly without the `encoding` keyword argument, so no decoding takes place.
> 
> **TL;DR:** The middleware should handle the decoding which the PR does.

Thanks for the clarification! I agree and as I said before, the suggested fix works perfectly fine 👍

**Comment by malefice** [comment]
Yeah, I just did the investigation and root cause analysis so that this can finally be resolved. I still very much prefer to have this fix upstream VS maintaining this (admittedly trivial) workaround in my projects.
